### PR TITLE
Add support for huge pages on Linux

### LINKED ISSUE #6351: Include hugepages metrics in metricbeat
If metricbeat could capture [hugepages metrics](https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt) such as `HugePages_Total`, `HugePages_Free`,`HugePages_Rsvd`,`HugePages_Surp`,`Hugepagesize`, and `Hugetlb` it would provide a more granular view of memory usage on the system.

`system.memory.used.pct` includes memory that the system has allocated to huge pages.
e.g.
```
[ hostname ] # cat /proc/meminfo | egrep "Total|Free|Huge"
MemTotal:       132027020 kB
MemFree:        10274084 kB
…
HugePages_Total:   56274
HugePages_Free:    23506
…
Hugepagesize:       2048 kB
```
Metricbeat reports this system as having ~92% memory used based on `100-(MemFree/(MemTotal/100))` - however there’s actually 23506*2048kB available as huge pages, which is about 58% of huge pages free.

### PR DESCRIPTION
**Pull request by jsoriano**
Add support for huge pages on Linux, fixes #6351, requires https://github.com/elastic/gosigar/pull/97

### CODE REVIEW

File: libbeat/metric/system/memory/memory.go [hunk 1]
```diff
@@ -87,3 +87,48 @@ func AddSwapPercentage(s *SwapStat) {
...
+	if s.Total == 0 {
+		return
+	}
+
+	perc := float64(s.Total-s.Free+s.Reserved) / float64(s.Total)
+	s.UsedPercent = common.Round(perc, common.DefaultDecimalPlacesCount)
+}
+
+// GetHugeTLBPagesEvent returns the event created from huge page usage metrics
+func GetHugeTLBPagesEvent(s *HugeTLBPagesStat) common.MapStr {
```
Comment (exekias) [inline libbeat/metric/system/memory/memory.go]: I think this function is not used?

**Comment by exekias** [comment]
Also have a look to system tests, you may want to check fields are in place in Linux: https://github.com/elastic/beats/blob/master/metricbeat/tests/system/test_system.py

**Comment by exekias** [comment]
Failing tests should be fixed by `make update`, that will format the code to the standard and update `NOTICE.txt`

File: libbeat/metric/system/memory/memory.go (hunk 1 above)
Comment (jsoriano) [reply libbeat/metric/system/memory/memory.go]: Umm, there are other similar `GetFooEvent` functions here, indeed they are not used in the beats repo, I though they were exported to be used somewhere else.

**Comment by exekias** [comment]
I saw https://github.com/elastic/gosigar/pull/97 was merged :tada:. Let's move this forward once we get a new tag version of it

**Comment by jsoriano** [comment]
gosigar 0.9.0 released, I will go on with this PR using this version.

File: libbeat/metric/system/memory/memory.go (hunk 1 above)
Comment (jsoriano) [reply libbeat/metric/system/memory/memory.go]: @exekias WDYT? Should we keep these `GetFooEvent` functions?

Comment (jsoriano) [reply libbeat/metric/system/memory/memory.go]: Removed new unused function, old ones are left there by now.

**Comment by ruflin** [comment]
Can you add a changelog entry?

**Comment by jsoriano** [comment]
Changelog entry added.
Please wait a bit before merging this, I'd like to try to reproduce this problem seen upgrading to gosigar 0.9.0: https://discuss.elastic.co/t/metricbeat-5-3-0-reporting-strange-memory-value/124825

**Comment by jsoriano** [comment]
We can continue merging this, it seems that the problem with gosigar was not introduced in 0.9.0, it was already reported before (#6271)

**Pull request merged**